### DISCUSSION TITLE
When a newer devices is not fully supported - how do we go about requesting it?

### DISCUSSION THREAD

**Discussion by mwareman** [fuatakgun/eufy_security#1092] [General]
I realize this is a combination effort between several contributors (and THANK YOU ALL!). This integration overall solves a massive gap with these cameras - lack of a local API. I really very much appreciate the effort and work that went into this overall.

I have a newer T8416 camera (the S350 Indoor Cam). It's fantastic - especially when paired with a Homebase 3 (and trained with several person recognitions). The mobile app tells me with good reliability WHO was seen. I would like to use this in Home Assistant automations. I also have the same limitation with the T8425 (Floodlight Cam E340)

When I look at a T8400 (Indoor Cam C120) or the T8410 (Indoor Cam E220) in Home Assistant - there are sensor attributes for 'Person Detected' 'Person Name'. I also have a lot of configuration options.

![image](https://github.com/fuatakgun/eufy_security/assets/25395017/d3397004-b273-4a13-b672-8401f3d39266)

On the T8416 (and T8425 - there is nothing. These attributes and configuration options do not exist. Camera events are not showing up in the logbook either. 

![image](https://github.com/fuatakgun/eufy_security/assets/25395017/0420c577-114c-4250-b9a6-c1155be98a8d)

This is odd - since the C120 and E330 do you (yet) support named person detection (but the integration already exposes this!) - but the S350 and E340 do (but the named person attribute is not exposed!).

https://support.eufy.com/s/article/Centralize-Your-Security-Ecosystem-with-HomeBase-3
![image](https://github.com/fuatakgun/eufy_security/assets/25395017/5a253dc8-086e-4fde-9b26-a05709a6e0fd)

I really have no idea where to ask for help. All the cameras are connected to a Homebase 3 if that makes any difference for AI enhancement and full time recording of my external facing cameras.

Why is it properties for a feature are showing on cameras where the feature is NOT supported - but not showing up on cameras where the feature IS supported! I would love there to be an easy fix for this...  but I have no idea where to start.

Thanks in advance for any suggestions.

**Comment by fuatakgun** [comment]
If add-on is not supporting the device, the functionality of it will be mostly msising.
https://bropat.github.io/eufy-security-client/#/supported_devices

If it is listed in this page but some functions are not working properly, we can check on integration end first. But, integration is mostly acting as a gateway with add-on and add-on takes care of communicating with devices, it is high probable that the specific functioanilty is not implemented on add-on side.

Integrationa asks add-on for properties and functions supported for each device and then creates related sensors and controls for them in home assisntant.

On the other hand, I know there device X can work fine over independent wifi but attaching it to homebase 3 might require extra work.

So, after reading this, let me know if you have further questions.
